launchpod/angular-launchpod/backend/src/test/java/com/google/launchpod/FileUploadServletTest.java

```diff
@@ -0,0 +1,666 @@
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.joda.time.DateTimeUtils;
+
+/**
+ * Runs unit tests for the FileUploadServlet that contains doPost() and doGet() methods.
+ */
+@RunWith(JUnit4.class)
+public class FileUploadServletTest extends Mockito {
+
+  @InjectMocks
+  private FileUploadServlet servlet = new FileUploadServlet();
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Rule // JUnit 4 uses Rules for testing specific messages
+  public ExpectedException thrown = ExpectedException.none();
+
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  // Keys
+  private static final String USER_FEED = "UserFeed";
+  private static final String PODCAST_TITLE = "title";
+  private static final String XML_STRING = "xmlString";
+  private static final String DESCRIPTION = "description";
+  private static final String LANGUAGE = "language";
+  private static final String TIMESTAMP = "timestamp";
+  private static final String EMAIL = "email";
+  private static final String MP3 = "mp3";
+  private static final String MP3_LINK = "mp3Link";
+  private static final String ID = "id";
+  private static final String ACTION = "action";
+
+  private static final String TEST_PODCAST_TITLE = "TEST_PODCAST_TITLE";
+  private static final String TEST_DESCRIPTION= "TEST_DESCRIPTION";
+  private static final String TEST_LANGUAGE= "en";
+  private static final long TEST_TIMESTAMP = System.currentTimeMillis();
+  private static final String TEST_ID = "123456";
+  private static final String TEST_ID_TWO = "789012";
+  private static final String TEST_PUBDATE = "2020/06/26 01:32:06";
+  private static final String TEST_EMAIL = "123@google.com";
+  private static final String TEST_INCORRECT_EMAIL = "123@cde.com";
+  private static final String TEST_EMAIL_TWO = "456@abc.com";
+  private static final String EMPTY_STRING = "";
+  private static final String TEST_XML_STRING = "test";
+  private static final String BASE_URL = "https://launchpod-step18-2020.appspot.com/rss-feed?id=";
+  private static final String GENERATE_XML_ACTION = "generateXml";
+  private static final String GENERATE_RSS_LINK = "generateRSSLink";
+  private static final SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss:SSS");
+
+  private static final String PROJECT_ID = "launchpod-step18-2020"; // The ID of your GCP project
+  private static final String BUCKET_NAME = "launchpod-mp3-files"; // The ID of the GCS bucket to upload to
+
+  @Before
+  public void setUp() throws ParseException {
+    MockitoAnnotations.initMocks(this);
+    Date fixedDateTime = dateFormatter.parse("07/29/2019 21:39:00:000");
+    DateTimeUtils.setCurrentMillisFixed(fixedDateTime.getTime());
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  private enum Action {
```

Comment:
reference the main code's Action instead of defining your own

Reply:
will be done in a separate PR specifically meant for common methods and constants